include/mbedtls/aes.h

```diff
@@ -69,26 +69,50 @@ extern "C" {
 // Regular implementation
 //
 
+/** VIA padlock needs 16 extra bytes for alignment. */
+#if (defined(MBEDTLS_PADLOCK_C) && defined(MBEDTLS_HAVE_X86)) ||        \
+    (defined(MBEDTLS_AESNI_C) && MBEDTLS_AESNI_HAVE_CODE == 2)
+#define MBEDTLS_PADLOCK_EXTRA 16
+#else
+#define MBEDTLS_PADLOCK_EXTRA 0
+#endif
+
+/** Each round key occupies 16 bytes. */
+#define MBEDTLS_AES_RK_UINT32S 16
+
+/** Define number of maximum rounds in AES key expansion.
+ *   See https://nvlpubs.nist.gov/nistpubs/FIPS/NIST.FIPS.197.pdf
+ *      - Section 5, Nr = Nk + 6
+ *      - Section 5.2, the length of round keys is Nb*(Nr+1)
+ */
+#if defined(MBEDTLS_AES_ONLY_128_BIT_KEY_LENGTH)
+#define MBEDTLS_AES_MAX_ROUNDS 10
+/** Allocate 16 extra bytes to store round key. */
+#define MBEDTLS_AES_RK_EXTRA (16 + MBEDTLS_PADLOCK_EXTRA)
+#else
+#define MBEDTLS_AES_MAX_ROUNDS 14
+/** Allocate 32 extra bytes for AES key expansion in the 192-bit and 256-bit
+    case in order to simplify key expansion. */
+#define MBEDTLS_AES_RK_EXTRA (32 + MBEDTLS_PADLOCK_EXTRA)
+#endif
+
 /**
  * \brief The AES context-type definition.
  */
 typedef struct mbedtls_aes_context {
     int MBEDTLS_PRIVATE(nr);                     /*!< The number of rounds. */
     size_t MBEDTLS_PRIVATE(rk_offset);           /*!< The offset in array elements to AES
                                                     round keys in the buffer. */
-#if defined(MBEDTLS_AES_ONLY_128_BIT_KEY_LENGTH) && !defined(MBEDTLS_PADLOCK_C)
-    uint32_t MBEDTLS_PRIVATE(buf)[44];           /*!< Aligned data buffer to hold
-                                                    10 round keys for 128-bit case. */
-#else
-    uint32_t MBEDTLS_PRIVATE(buf)[68];           /*!< Unaligned data buffer. This buffer can
-                                                    hold 32 extra Bytes, which can be used for
-                                                    one of the following purposes:
-                                                    <ul><li>Alignment if VIA padlock is
-                                                    used.</li>
-                                                    <li>Simplifying key expansion in the 256-bit
-                                                    case by generating an extra round key.
-                                                    </li></ul> */
-#endif /* MBEDTLS_AES_ONLY_128_BIT_KEY_LENGTH && !MBEDTLS_PADLOCK_C */
+    /** Data buffer to hold round keys. This buffer can hold 16 / 32 / 48
+     *  extra bytes, which can be used for one of the following purposes:
+     *   - Extra space to store round key.
+     *   - Alignment if VIA padlock is used.
+     *   - Simplifying key expansion in the 192-bit and 256-bit case by
+     *   generating an extra round key.
+     */
+    uint32_t MBEDTLS_PRIVATE(buf)
+    [(MBEDTLS_AES_MAX_ROUNDS * MBEDTLS_AES_RK_UINT32S + MBEDTLS_AES_RK_EXTRA) /
```

Comment:
Would it be more readable if we abstract the calculation to a new macro like
`MBEDTLS_AES_BUF_LEN`?